src/custom-actions.ts

```diff
@@ -0,0 +1,222 @@
+import { CustomAction, CustomActionsPosition, CustomActionTarget } from './types';
+import { arrayIncludesString } from './utils';
+
+export interface CustomActionsValidationResult {
+    actions: CustomAction[];
+    errors: string[];
+}
+
+type CustomActionValidation = {
+    isValid: boolean;
+    reason: string;
+};
+
+/**
+ * This class encapsulates the logic for validating custom actions.
+ * It provides a clean interface for validating custom actions with proper error handling
+ * and validation rules.
+ * 
+ * @hidden
+ */
+export class CustomActions {
```

Comment:
Every method here is a static function, there isn't any concept of class and instance creation needed here.
You can just create those functions in this file and directly export them